### PR TITLE
Support OCaml 4.08+

### LINKED ISSUE #22: OCaml 4.08+ support
Thanks for getting this project moving. Right now it doesn't seem like it's possible to depend on this package in newer versions of OCaml. I had to add the following to my esy resolutions:

```json
    "@baransu/graphql_ppx_re": {
      "source": "baransu/graphql_ppx_re:esy.json#c7fab6c",
      "override": {
        "build": "dune build -p graphql_ppx",
        "dependencies": {
          "@opam/ocaml-migrate-parsetree": "*",
          "@esy-ocaml/reason": "*",
          "refmterr": null
        }
      }
    }
```

I think these could get integrated in master. I don't know what refmterr is but it just doesn't build on OCaml 4.09 which is what I was testing, and removing it works fine. The `esy.json` also looks weird, and many of its dependencies are _very_ outdated (e.g. Menhir, OMP). I can submit an enhancement PR if you want.

### PR DESCRIPTION
**Pull request by baransu**
This replaces #23 and fixes #22 

We now have separate files for BuckleScript 5, BuckleScript 6+ and Native with latest OCaml.

/ cc @anmonteiro

### CODE REVIEW

**Comment by anmonteiro** [comment]
@baransu is there anything holding this PR back? the build failure on windows looks related to creating a directory that already exists. Can that be created conditionally?

**Comment by baransu** [comment]
I'm not sure if I'm responsible for creating this directory. It's not directly `.graphql_ppx_cache` but some `esy` directory related to this cache folder. Not sure what's going on TBH.

**Comment by anmonteiro** [comment]
Esy doesn't create paths within the sandbox. I'd try `esy dune runtest` instead of `esy b dune runtest`

File: .github/workflows/pipeline.yml [hunk 1]
```diff
@@ -23,29 +23,33 @@ jobs:
...
+
+      # OCaml 4.09
+      - name: native
         run: |
-          esy install 
+          esy install
           esy b
-      - name: test-native
-        run: |
           esy b dune runtest -f
```
Comment (anmonteiro) [inline .github/workflows/pipeline.yml]:
```suggestion
          esy dune runtest -f
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -23,37 +23,42 @@ jobs:
...
+      # OCaml 4.09
+      - name: native
         run: |
           esy install
           esy b
-      - name: test-native
-        run: |
-          esy b dune runtest -f
-      - name: test-bsb6
+          esy B dune runtest -f
```
Comment (anmonteiro) [inline .github/workflows/release.yml]:
```suggestion
          esy dune runtest -f
```

**Comment by jfrolich** [comment]
@baransu: Can you look at the pipeline.yml file, there is something not right there (creating binaries for bsb6 twice).

**Comment by baransu** [comment]
Should be fixed. Thanks @jfrolich for attention!

**Pull request merged**